### ISSUE TITLE
Fast tokenizers padding and prefix space

### ISSUE THREAD

**Issue by fte10kso** [huggingface/transformers#2947]
# 🐛 Bug
`pad_to_max_length=True` Doesn't seem to do anything when using fast tokenizers.

```python
tokenizer_roberta = RobertaTokenizer.from_pretrained('roberta-base')
tokenizer_roberta_fast = RobertaTokenizerFast.from_pretrained('roberta-base')

tokenizer_bert = BertTokenizer.from_pretrained('bert-base-uncased')
tokenizer_bert_fast = BertTokenizerFast.from_pretrained('bert-base-uncased')

def test_encode_decode(name, text, tokenizer):
    encoded = tokenizer.encode(text, max_length=10, pad_to_max_length=True)
    decoded = tokenizer.decode(encoded)
    print(name)
    print(encoded)
    print(decoded)
    print()
    
text = 'hello huggingface'

test_encode_decode('bert', text, tokenizer_bert)
test_encode_decode('bert_fast', text, tokenizer_bert_fast)
test_encode_decode('roberta', text, tokenizer_roberta)
test_encode_decode('roberta_fast', text, tokenizer_roberta_fast)
```

**Output:**
```
bert
[101, 7592, 17662, 12172, 102, 0, 0, 0, 0, 0]
[CLS] hello huggingface [SEP] [PAD] [PAD] [PAD] [PAD] [PAD]

bert_fast
[101, 7592, 17662, 12172, 102]
[CLS] hello huggingface [SEP]

roberta
[0, 20760, 31164, 9021, 2, 1, 1, 1, 1, 1]
<s> hello huggingface</s><pad><pad><pad><pad><pad>

roberta_fast
[0, 42891, 31164, 9021, 2]
<s>hello huggingface</s>
```
Additionally I can't seem to make `add_prefix_space=True` to work with `RobertaTokenizerFast`. I can only get the same output as `RobertaTokenizer` if i manually prepend a space. I saw the warning and link to #2778 but I'm not sure if I follow completely.

Thanks for taking the time!

Edit: I'm up to date with master, latest commit 53ce3854a16ad2a715bc6ac8af3e30c18b5a1d11

**Comment by mfuntowicz** [comment]
This should be fixed in master on commit cc6775cdf5b20ad382613d3bdbf0dd8364d23219.

If you want to give it a try, otherwise the first maintenance release will soon be live.

**Comment by fte10kso** [comment]
That fixed it for me, thank you!

Also figured out what i was doing wrong with `add_prefix_space`. Instead of passing it to encode as for the regular tokenizers it should be provided at init e.g.

```python
from transformers import RobertaTokenizerFast
t = RobertaTokenizerFast.from_pretrained('roberta-base', add_prefix_space=True)
t.decode(t.encode('hello huggingface'))
# '<s> hello huggingface</s>'
```
